uptakeprojection/R/get_nis_2022.R

```diff
@@ -0,0 +1,197 @@
+        sd_maybe = product_sd(estimate_primary, estimate_maybe, sd_primary, sd_maybe),
+        estimate_yes = estimate_yes * estimate_primary,
+        sd_yes = product_sd(estimate_primary, estimate_yes, sd_primary, sd_yes),
+        estimate_no = estimate_no * estimate_primary + (1 - estimate_primary),
+        sd_no = product_sd(estimate_primary, estimate_no, sd_primary, sd_no)
+      ) |>
+      tidyr::pivot_longer(c(starts_with("estimate")),
+        names_to = "indicator_category",
+        values_to = "estimate"
+      ) |>
+      tidyr::pivot_longer(c(starts_with("sd")),
+        names_to = "indicator_category_sd",
+        values_to = "sd"
+      ) |>
+      mutate(
+        indicator_category = stringr::str_split_i(indicator_category, pattern = "_", i = 2),
+        indicator_category_sd = stringr::str_split_i(indicator_category_sd,
+          pattern = "_", i = 2
+        )
+      ) |>
+      filter(
+        indicator_category == indicator_category_sd,
+        indicator_category != "primary",
+        !is.na(estimate)
+      ) |>
+      # convert back to percents
+      mutate(across(c(estimate, sd), function(x) x * 100)) %>%
+      select(
+        time_type, date, indicator, indicator_category, geography, group_name,
+        group_category, estimate, sd, season
+      )
+  }
+
+  saved <- data %>%
+    arrange(date) %>%
+    filter(!group_category %in% c(
+      "18 – 49 years", "60+ years", "65+ years"
+    )) %>%
+    mutate(
+      group_category = case_when(
+        group_category == "18 – 29 years" ~ "18-29 years",
+        group_category == "30 – 39 years" ~ "30-39 years",
+        group_category == "40 – 49 years" ~ "40-49 years",
+        group_category == "50 – 64 years" ~ "50-64 years",
+        group_category == "65 – 74 years" ~ "65-74 years",
+        group_category == "American Indian/Alaska Native, non-Hispanic" ~
+          "American Indian/Alaska Native, Non-Hispanic",
+        group_category == "Asian, non-Hispanic" ~ "Asian, Non-Hispanic",
+        group_category == "Black, non-Hispanic" ~ "Black, Non-Hispanic",
+        group_category == "Other or multiple races, non-Hispanic" ~ "Other, Non-Hispanic",
+        group_category == "Native Hawaiian/Pacific Islander, non-Hispanic" ~
+          "Pacific Islander/Native Hawaiian, Non-Hispanic",
+        group_category == "White, non-Hispanic" ~ "White, Non-Hispanic",
+        group_category == "Above poverty, income <$75k" ~ "Above Poverty, Income < $75k",
+        group_category == "Above poverty, income >=$75k" ~ "Above Poverty, Income >= $75k",
+        group_category == "Below poverty" ~ "Below Poverty",
+        group_category == "Unknown income" ~ "Poverty Status Unknown",
+        group_category == "All adults age 18+ years" ~ "Overall",
+        group_category == "18+ years" ~ "Overall",
+        TRUE ~ group_category
+      ),
+      group_name = ifelse(group_name == "Race/Ethnicity (7 level)", "Race/Ethnicity", group_name),
```

Comment:
this seems like another series of places where `case_match()` would clarify what's going on